### PR TITLE
TYP: series apply method adds type hints

### PR DESCRIPTION
**Pull request by aniaan**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -3979,7 +3979,9 @@ def transform(
     ) -> FrameOrSeriesUnion:
         return transform(self, func, axis, *args, **kwargs)
 
-    def apply(self, func, convert_dtype=True, args=(), **kwds):
+    def apply(
+        self, func: AggFuncType, convert_dtype: bool = True, args: tuple = (), **kwds
+    ):
```
Comment (simonjayhawkins) [inline pandas/core/series.py:3984]:
```suggestion
    ) -> FrameOrSeriesUnion:
```

File: pandas/core/series.py [hunk 2]
```diff
@@ -3979,7 +3979,9 @@ def transform(
     ) -> FrameOrSeriesUnion:
         return transform(self, func, axis, *args, **kwargs)
 
-    def apply(self, func, convert_dtype=True, args=(), **kwds):
+    def apply(
+        self, func: AggFuncType, convert_dtype: bool = True, args: tuple = (), **kwds
```
Comment (simonjayhawkins) [inline pandas/core/series.py:3983]: use Tuple from typing and add type parameters where possible

```suggestion
        self, func: AggFuncType, convert_dtype: bool = True, args: Tuple[Any] = (), **kwds
```

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @BeanNan for the PR. a couple of suggestions.

**Comment by BeanNan** [comment]
> Thanks @BeanNan for the PR. a couple of suggestions.

Corrected, please take a look

**Comment by simonjayhawkins** [comment]
see https://github.com/pandas-dev/pandas/runs/1673701200 for mypy failures

**Comment by BeanNan** [comment]
CI has passed, please take a look

**Comment by jreback** [comment]
can you merge master; your changes will now work in pandas/core/apply.py where thing were moved.

**Comment by BeanNan** [comment]
Updated, but there was a slight problem with the CI that doesn't seem to be related to this commit

**Comment by simonjayhawkins** [comment]
> Updated, but there was a slight problem with the CI that doesn't seem to be related to this commit

yep, unrelated.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @BeanNan lgtm

FYI, the casts would be unnecessary with @overload, but OK with gradual typing to do things in small incremental steps.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Pull request merged**